timesketch/lib/analyzers/crash.py

```diff
@@ -0,0 +1,178 @@
+                'event_identifier:"1"',
+                'event_level:"2"', # Level: Error
+            ),
+            'Event - .NET App Crash' : (
+                'data_type:"windows:evtx:record"',
+                'source_name:".NET Runtime"',
+                'event_identifier:"1026"',
+                'event_level:"2"', # Level: Error
+            ),
+            'File - WER Report' : (
+                'data_type:"fs:stat"',
+                'filename:"/Microsoft/Windows/WER/"',
+                'filename:/((Non)?Critical|AppCrash)_.*/',
+                'file_entry_type:("directory" or "2")',
+            ),
+            'Registry - Crash Reporting' : (
+                'data_type:"windows:registry:key_value"',
+                r'key_path:r"\\Control\\CrashControl"',
+                'values:("LogEvent: REG_DWORD_LE 0" OR "SendAlert: REG_DWORD_LE 0" OR "CrashDumpEnabled: REG_DWORD_LE 0")',
+            ),
+            'Registry - Error Reporting' : (
+                'data_type:"windows:registry:key_value"',
+                r'key_path:"\\Software\\Microsoft\\PCHealth\\ErrorReporting"',
+                'values:("DoReport: REG_DWORD_LE 0" OR "ShowUI: REG_DWORD_LE 0")',
+            ),
+        }
+
+        # Creator of nested Elastic Search queries.
+        def formulate_query(elements):
+            conditions = []
+            for e in elements.values():
+                conditions += ['({0})'.format(' AND '.join(e))]
+            return ' OR '.join(conditions)
+
+        # Re-usable filename matching function.
+        def extract_filename(text, regex):
+            if '.exe' in text.lower():
+                match = regex.search(message)
+                if match:
+                    # The regex can match on full file paths and filenames,
+                    # so only return the filename.
+                    return min([m for m in match.groups() if m])
+            return None
+
+        # For adding information to identified entries.
+        def mark_as_crash(event, filename):
+            event.add_star()
+            event.add_attributes({'crash_app': filename})
+            event.add_label('crash')
+
+        query = formulate_query(query_elements)
+
+        return_fields = ['data_type', 'message', 'filename']
+
+        # Generator of events based on your query.
+        events = self.event_stream(
+            query_string=query, return_fields=return_fields)
+
+        # Get the executable filenames.
+        # Examples: 
+        #   "[...]\WER\ReportQueue\AppCrash_notepad.exe_8d163e29d3960561ca2e9723640cd8fff5c2ad5_cab_0a5810ac"
+        #   "[...]/WER/ReportQueue/NonCritical_x64_d473a376adfb18a7b165c5e3c26de43cd8bccb_cab_07fd2620"
+        #   "Strings: ['iexplore.exe', '8.0.7601.17514' [...]"
+        re_filename = re.compile(r'(?:\\|\/)(?:AppCrash|Critical|NonCritical)_(.+?\.exe)_[a-f0-9]{16,}|\'([^\']+\.exe)\'', re.I)
+
+        # Container for filenames of crashed applications.
+        filenames = dict()
```

Comment:
why is this a dict? you don't seem to do anything with the actual values, this is all just

filenames[filename] = ''

so why not have this as

filenames = set()

or something like that?